src/huts/repositorySpec.js

```diff
@@ -1,29 +1,30 @@
 const HutsRepository = require("./repository");
 const { sqlEquality } = require("../../spec/support/customEqualityTesters");
+const mockFailOnQuery = require("../../spec/support/mockFailOnQuery");
 
 describe("huts repository" , function() {
+    let db;
 
     beforeEach(function () {
-        jasmine.addCustomEqualityTester(sqlEquality);
+        db = {
+            query: jasmine.createSpy("db.query")
+        };
+        jasmine.addCustomEqualityTester(sqlEquality); 
     });
 
     describe("constructor function", function() {
-        it("creates a repository if the huts and role_connections table already exists",async function() {
-            let db = {
-                query: async (query) => {
-                    if(query == "SELECT 'public.huts'::regclass") return;
-                    if(query == "SELECT 'public.role_connections'::regclass") return;
-                }
-            };
 
-            let actualError = null;
-            let hutsRepository = new HutsRepository(db);
+        beforeEach(function() {
+            db.query.and.callFake(async (query) => {
+                if(query == "SELECT 'public.huts'::regclass") return;
+                if(query == "SELECT 'public.role_connections'::regclass") return;
+            })
+        });
 
-            try {
-                await hutsRepository.initialize();
-            } catch(error) {
-                actualError = error;
-            }
+        it("creates a repository if the huts and role_connections table already exists",async function() {
+            let hutsRepository = new HutsRepository(db);
+            
+            let actualError = await getErrorsFromRunningFunction(hutsRepository.initialize);
```

Comment:
In JavaScript, passing `hutsRepository.initialize` means passing just the function, stripping it of context.

In this case it makes no difference, but technically hutsRepository.initialize might reference `this` internally, which would only work while it is on `hutsRepository`. Passing it as you do here, strips that context from it.

For this reason, I think it might be nicer like this:

```js
let actualError = await getErrorsFromRunningFunction(() => hutsRepository.initialize());
```

This means that `initialize` might still use `this`. (I know it doesn't in this case, but always doing the same thing seems nicer to me.)

Reply:
great learning, thank you!